curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md

```diff
@@ -1,28 +1,33 @@
 ---
 id: 657e2ecee9f60092c89338d9
-title: Step 84
+title: Step 85
 challengeType: 0
-dashedName: step-84
+dashedName: step-85
 ---
 
 # --description--
 
-If the user rolls a small straight, then the `Small straight` radio button option should be enabled with a score of `30` points.
+If the user rolls a small straight, then the `Small straight` radio button option should be enabled with a score of `30` points. A small straight consists of four separate dice rolls arranged in numerical order. 
 
-Start by creating an `if` statement to check if `uniqueNumbersStr` is included in the `smallStraightsArr`. If so, call the `updateRadioOption` with `3` and `30` for the arguments. 
+To get started, iterate through `smallStraightsArr` using any loop you prefer. If using a higher-order function like `forEach` or `some`, the callback function should take an argument of `straight`. If using `for...of`, the loop variable should also be `straight`.
+
+Inside the loop,  if `uniqueNumbersStr` has a combination that matches a small straight, call the `updateRadioOption` with `3` and `30` as the arguments. 
 
 # --hints--
 
-You should use the `includes` method to check if `uniqueNumbersStr` is included in the `smallStraightsArr`.
+You should use the `includes` method to check if `straight` is included in the `uniqueNumbersStr`. 
 
 ```js
-assert.match(code, /if\s*\(\s*smallStraightsArr\.includes\(\s*uniqueNumbersStr\s*\)\s*\)\s*{?/);
+assert.match(code, /\s*i?f?\(?\s*uniqueNumbersStr\.includes\(\s*(straight|smallStraightsArr\[i\])\s*\)\)\s*\)?\s*\{/)  
```

Comment:
That is because there are two different equally valid solutions to this challenge.

Solution 1:

`
 if (smallStraightsArr.some(straight => uniqueNumbersStr.includes(straight))) {
    updateRadioOption(3, 30);
  }
`

Solution 2:

`
smallStraightArr.forEach(straight => {
  if(uniqueNumbersStr.includes(straight)) {
    updateRadioOption(3, 30);
  }
}
`

If I remove the question marks, then only one of these solutions become viable.